e2etest/e2e_test.go

```diff
@@ -0,0 +1,205 @@
+	// create event consumer
+	queueCfg := config.DefaultQueueConfig()
+	queueConsumer, err := setupTestQueueConsumer(t, queueCfg)
+	require.NoError(t, err)
+	stakingChan, err := queueConsumer.ActiveStakingQueue.ReceiveMessages()
+	require.NoError(t, err)
+
+	defer func() {
+		err := queueConsumer.Stop()
+		require.NoError(t, err)
+	}()
+
+	n := 1
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	stakingEventList := make([]*queuecli.StakingEvent, 0)
+	for i := 0; i < n; i++ {
+		stakingEvent := queuecli.NewActiveStakingEvent(
+			hex.EncodeToString(bbndatagen.GenRandomByteArray(r, 10)),
+			hex.EncodeToString(bbndatagen.GenRandomByteArray(r, 10)),
+			[]string{hex.EncodeToString(bbndatagen.GenRandomByteArray(r, 10))},
+			1000,
+		)
+		err = queueConsumer.PushActiveStakingEvent(&stakingEvent)
+		require.NoError(t, err)
+		stakingEventList = append(stakingEventList, &stakingEvent)
+	}
+
+	for i := 0; i < n; i++ {
+		stakingEventBytes := <-stakingChan
+		var receivedStakingEvent queuecli.StakingEvent
+		err = json.Unmarshal([]byte(stakingEventBytes.Body), &receivedStakingEvent)
+		require.NoError(t, err)
+		require.Equal(t, stakingEventList[i].StakingTxHashHex, receivedStakingEvent.StakingTxHashHex)
+		err = queueConsumer.ActiveStakingQueue.DeleteMessage(stakingEventBytes.Receipt)
+		require.NoError(t, err)
+	}
+}
+
+// TestStakingLifecycle verifies the happy path of BTC delegation state transitions
+// PENDING -> VERIFIED -> ACTIVE -> UNBONDING/EARLY_UNBONDING
+// 1. Create BTC delegation without inclusion proof in Babylon node (pre-approval flow)
+// 2. Wait for delegation to be PENDING in Indexer DB
+// 3. Generate and insert new covenant signature in Babylon node
+// 4. Wait for delegation to be VERIFIED in Indexer DB
+// 5. Submit inclusion proof to Babylon node
+// 6. Wait for delegation to be ACTIVE in Babylon node
+// 7. Wait for delegation to be ACTIVE in Indexer DB
+// 8. Verify active staking event emitted by Indexer
+// 9. Early unbonding on Babylon node
+// 10. Wait for delegation to be UNBONDED in Babylon node
+// 11. Wait for delegation to be UNBONDING and sub-state to be EARLY_UNBONDING in Indexer DB
+// 12. Verify unbonding staking event emitted by Indexer
+func TestStakingLifecycle(t *testing.T) {
+	// Segw is activated at height 300. It's necessary for staking/slashing tx
+	numMatureOutputs := uint32(300)
+	ctx := context.Background()
+
+	tm := StartManager(t, numMatureOutputs, defaultEpochInterval)
+	defer tm.Stop(t)
+
+	// Insert all existing BTC headers to babylon node
+	tm.CatchUpBTCLightClient(t)
+
+	// Create finality provider in Babylon node
+	fpPK, fpSK := tm.CreateFinalityProvider(t)
+
+	// Wait for finality provider to be stored in Indexer DB
+	tm.WaitForFinalityProviderStored(t, ctx, fpPK.BtcPk.MarshalHex())
+
+	// Create BTC delegation without inclusion proof in Babylon node
+	stakingMsgTx, stakingSlashingInfo, unbondingSlashingInfo, _ := tm.CreateBTCDelegationWithoutIncl(t, fpSK)
```

Comment:
Given that this change for expiry checker is for phase-1 transactions, we should test the case where inclusion proof is included in eoi